protocol/shannon/protocol.go

```diff
@@ -0,0 +1,176 @@
+package shannon
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/pokt-network/poktroll/pkg/polylog"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+	sdk "github.com/pokt-network/shannon-sdk"
+
+	"github.com/buildwithgrove/path/gateway"
+	"github.com/buildwithgrove/path/protocol"
+)
+
+// gateway package's Protocol interface is fulfilled by the Protocol struct
+// below using methods that are specific to Shannon.
+var _ gateway.Protocol = &Protocol{}
+
+// FullNode defines the set of capabilities the Shannon protocol integration needs
+// from a fullnode for sending relays.
+type FullNode interface {
+	GetServiceApps(protocol.ServiceID) ([]apptypes.Application, error)
+	// Note: Shannon returns the latest session for a service+app combination if no blockHeight is provided.
+	// This is used here because the gateway only needs the current session for any service+app combination.
```

Comment:
`This is used here because` is not a very "evergreen" comment for an interface. Please reword.